src/lookup.rs

```diff
@@ -436,6 +474,101 @@ pub(super) async fn fetch_channel_updates<L: Deref>(delta_set: &mut DeltaSet, cl
 	log_info!(logger, "Processed intermediate rows ({}) (delta size: {}): {:?}", intermediate_update_count, delta_set.len(), start.elapsed());
 }
 
+pub(super) async fn fetch_node_updates<L: Deref>(delta_set: &mut NodeDeltaSet, client: &Client, last_sync_timestamp: u32, logger: L) where L::Target: Logger {
+	let start = Instant::now();
+	let last_sync_timestamp_float = last_sync_timestamp as f64;
+
+	// get the latest node updates prior to last_sync_timestamp
+	let reference_rows = client.query_raw("
+		SELECT DISTINCT ON (public_key) public_key, CAST(EXTRACT('epoch' from seen) AS BIGINT) AS seen, announcement_signed FROM node_announcements
+		WHERE seen < TO_TIMESTAMP($1)
+		ORDER BY public_key ASC, seen DESC
+		", [last_sync_timestamp_float]).await.unwrap();
+	let mut pinned_rows = Box::pin(reference_rows);
+
+	log_info!(logger, "Fetched node announcement reference rows in {:?}", start.elapsed());
+
+	let mut reference_row_count = 0;
+
+	while let Some(row_res) = pinned_rows.next().await {
+		let current_reference = row_res.unwrap();
+
+		let seen = current_reference.get::<_, i64>("seen") as u32;
+		let blob: Vec<u8> = current_reference.get("announcement_signed");
+		let mut readable = Cursor::new(blob);
+		let unsigned_node_announcement = NodeAnnouncement::read(&mut readable).unwrap().contents;
+		let node_id = unsigned_node_announcement.node_id;
+
+		let current_node_delta = delta_set.entry(node_id).or_insert(NodeDelta::default());
+		let address_set: HashSet<SocketAddress> = unsigned_node_announcement.addresses.into_iter().collect();
+		(*current_node_delta).last_details_before_seen.get_or_insert(NodeDetails {
+			seen,
+			features: unsigned_node_announcement.features,
+			addresses: address_set
+		});
```

Comment:
You can avoid unnecessarily allocating a `HashSet` by using `Option::get_or_insert_with`. Only necessary if a non-empty `delta_set` can be passed in, though.